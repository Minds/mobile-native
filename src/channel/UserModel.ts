import { observable, action, runInAction } from 'mobx';

import { MINDS_CDN_URI, GOOGLE_PLAY_STORE } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import ChannelService from './ChannelService';
import sessionService from '../common/services/session.service';
import apiService from '../common/services/api.service';
import logService from '../common/services/log.service';
import { SupportTiersType } from '../wire/WireTypes';

//@ts-nocheck
export const USER_MODE_OPEN = 0;
export const USER_MODE_MODERATED = 1;
export const USER_MODE_CLOSED = 2;

/**
 * User model
 */
export default class UserModel extends BaseModel {
  programs;
  merchant;
  /**
   * Eth wallet
   */
  eth_wallet: string = '';
  disable_autoplay_videos?: boolean;
  sums;
  btc_address?: string;
  icontime!: string;
  username!: string;
  briefdescription!: string;
  city!: string;
  name!: string;
  is_admin = false;
  plus: boolean = false;
  verified: boolean = false;
  founder: boolean = false;
  rewards: boolean = false;
  last_accepted_tos: number = 0;
  subscriptions_count: number = 0;
  carousels?: Array<any>;
  dob?: string;

  tags: Array<string> = [];
  groupsCount: number = 0;

  /**
   * @var {boolean}
   */
  @observable blocked;

  /**
   * @var {number}
   */
  @observable subscribers_count;

  /**
   * @var {number}
   */
  @observable impressions;

  /**
   * @var {boolean}
   */
  @observable subscribed;

  /**
   * @var {boolean}
   */
  @observable mature_visibility = false;

  /**
   * @var {boolean}
   */
  @observable pending_subscribe = false;

  /**
   * @var {numeric}
   */
  @observable mode = 0;

  /**
   * @var {boolean}
   */
  @observable email_confirmed = false;

  @observable wire_rewards;

  @observable pro: boolean = false;

  onchain_booster: number = 0;

  /**
   * Confirm email
   * @param {Object} params
   */
  confirmEmail = async (params) => {
    // call any api endpoint with the param
    try {
      await apiService.get('api/v2/entities/', { urn: this.urn, ...params });
      this.setEmailConfirmed(true);
      return true;
    } catch (error) {
      return false;
    }
  };

  /**
   * Get the user icon time
   */
  getOwnerIcontime(): string {
    if (sessionService.getUser().guid === this.guid) {
      return sessionService.getUser().icontime;
    } else {
      return this.icontime;
    }
  }

  @action
  toggleMatureVisibility() {
    if (GOOGLE_PLAY_STORE) return;
    this.mature_visibility = !this.mature_visibility;
  }

  @action
  toggleSubscription = async () => {
    const value = !this.subscribed;
    this.subscribed = value;

    try {
      const metadata = this.getClientMetadata();
      await ChannelService.toggleSubscription(this.guid, value, metadata);
      UserModel.events.emit('toggleSubscription', this);
    } catch (err) {
      runInAction(() => {
        this.subscribed = !value;
      });
      throw err;
    }
  };

  @action
  async toggleBlock(value: boolean | null = null) {
    value = value === null ? !this.blocked : value;

    try {
      await ChannelService.toggleBlock(this.guid, value);
      this.blocked = value;
    } catch (err) {
      this.blocked = !value;
      logService.exception('[ChannelStore] toggleBlock', err);
    }
  }

  @action
  setMode(value) {
    this.mode = value;
  }

  @action
  setEmailConfirmed(value) {
    this.email_confirmed = value;
  }

  @action
  setTier(tier: SupportTiersType, type: 'usd' | 'tokens') {
    const wire_rewards = this.wire_rewards;
    if (!wire_rewards.rewards) {
      wire_rewards.rewards = {
        money: [] as SupportTiersType[],
        tokens: [] as SupportTiersType[],
      };
    }
    if (type === 'tokens') {
      if (!wire_rewards.rewards.tokens) {
        wire_rewards.rewards.tokens = [] as SupportTiersType[];
      }
      wire_rewards.rewards.tokens.push(tier);
    } else {
      if (!wire_rewards.rewards.money) {
        wire_rewards.rewards.money = [] as SupportTiersType[];
      }
      wire_rewards.rewards.money.push(tier);
    }
    this.wire_rewards = wire_rewards;
  }

  @action
  togglePro() {
    this.pro = !this.pro;
  }

  /**
   * Is admin
   */
  isAdmin() {
    return this.is_admin;
  }

  /**
   * current user is owner of the channel
   */
  isOwner = () => {
    return sessionService.getUser().guid === this.guid;
  };

  /**
   * Get banner source
   */
  getBannerSource() {
    if (this.carousels) {
      return {
        uri: this.carousels[0].src,
      };
    }
    return {
      uri: `${MINDS_CDN_URI}fs/v1/banners/${this.guid}/fat/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Get avatar source
   * @param {string} size
   */
  getAvatarSource(size = 'medium') {
    return {
      uri: `${MINDS_CDN_URI}icon/${this.guid}/${size}/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Has banner
   * @returns {boolean}
   */
  hasBanner() {
    return !!this.carousels;
  }

  /**
   * Is closed
   * @returns {boolean}
   */
  isClosed() {
    return this.mode === USER_MODE_CLOSED;
  }

  /**
   * Is open
   * @returns {boolean}
   */
  isOpen() {
    return this.mode === USER_MODE_OPEN;
  }

  /**
   * Is moderated
   * @returns {boolean}
   */
  isModerated() {
    return this.mode === USER_MODE_MODERATED;
  }

  /**
   * Is subscribed
   * @returns {boolean}
   */
  isSubscribed() {
    return !!this.subscribed;
  }

  /**
   * Has avatar
   */
  hasAvatar(): boolean {
    return this.icontime !== this.time_created;
  }

  /**
   * Request subscribe
   */
  async subscribeRequest() {
    if (this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = true;
      await apiService.put(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = false;
      logService.exception(err);
    }
  }

  /**
   * Cancel subscribe request
   */
  async cancelSubscribeRequest() {
    if (!this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = false;
      await apiService.delete(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = true;
      logService.exception(err);
    }
  }
}
