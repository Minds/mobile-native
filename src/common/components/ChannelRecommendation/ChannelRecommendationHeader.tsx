import { useNavigation } from '@react-navigation/native';
import { observer } from 'mobx-react-lite';
import React from 'react';
import { View } from 'react-native';
import FeedListInvisibleHeader from '~/common/components/FeedListInvisibleHeader';
import MenuSheet from '~/common/components/bottom-sheet/MenuSheet';
import { useLegacyStores } from '~/common/hooks/use-stores';
import i18nService from '~/common/services/i18n.service';
import { B2, H4, Icon, Row } from '~/common/ui';
import ThemedStyles from '~/styles/ThemedStyles';
import { styles as headerStyles } from '~/topbar/Topbar';
import { ChannelRecommendationStore } from './hooks/useChannelRecommendation';
import useChannelRecommendationContext from './hooks/useChannelRecommendationContext';

type PropsType = {
  location: string;
  shadow?: boolean;
  recommendationStore?: ChannelRecommendationStore;
};

function ChannelRecommendationHeader({
  location,
  shadow,
  recommendationStore,
}: PropsType) {
  const navigation = useNavigation();
  const { dismissal } = useLegacyStores();
  const dismissible = location !== 'channel';
  const recommendation =
    useChannelRecommendationContext() || recommendationStore;

  const shouldRender =
    Boolean(recommendation?.result?.entities.length) &&
    (dismissible
      ? !dismissal.isDismissed('channel-recommendation:feed')
      : true);

  const sheetOptions = React.useMemo(
    () => [
      {
        title: i18nService.t('removeFromFeed'),
        onPress: () => dismissal.dismiss('channel-recommendation:feed'),
        iconName: 'close',
        iconType: 'material-community',
      },
    ],
    [dismissal],
  );

  const { alignSelfCenterMaxWidth, bgPrimaryBackground } = ThemedStyles.style;

  return shouldRender ? (
    <View
      style={[
        alignSelfCenterMaxWidth,
        bgPrimaryBackground,
        shadow ? headerStyles.shadow : undefined,
      ]}>
      <Row align="centerBetween" vertical="L" horizontal="L">
        <H4>{i18nService.t('recommendedForYou')}</H4>
        <Row align="centerBoth">
          <B2
            color="link"
            onPress={() => navigation.navigate('SuggestedChannel')}>
            {i18nService.t('seeMore')}
          </B2>

          {dismissible && (
            <MenuSheet items={sheetOptions}>
              <Icon name="more" size="large" left="M" />
            </MenuSheet>
          )}
        </Row>
      </Row>
    </View>
  ) : (
    <FeedListInvisibleHeader />
  );
}

export default observer(ChannelRecommendationHeader);
