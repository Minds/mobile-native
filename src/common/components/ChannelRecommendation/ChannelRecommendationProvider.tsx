import React, { createContext } from 'react';
import type UserModel from '~/channel/UserModel';
import {
  ChannelRecommendationStore,
  useChannelRecommendation,
} from './hooks/useChannelRecommendation';

export type ChannelRecommendationContextType =
  | ChannelRecommendationStore
  | undefined;

export const ChannelRecommendationContext = createContext<ChannelRecommendationContextType>(
  undefined,
);

type ProviderPropTypes = {
  children: React.ReactNode;
  location: string;
  channel?: UserModel;
};

export const ChannelRecommendationProvider = ({
  children,
  location,
  channel,
}: ProviderPropTypes) => {
  const channelRecommendations = useChannelRecommendation(location, channel);
  return (
    <ChannelRecommendationContext.Provider value={channelRecommendations}>
      {children}
    </ChannelRecommendationContext.Provider>
  );
};
