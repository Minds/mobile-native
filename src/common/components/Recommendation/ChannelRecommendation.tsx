import React, { FC } from 'react';
import RecommendationBody, {
  RecommendationBodyProps,
} from './RecommendationBody';
import RecommendationHeader from './RecommendationHeader';
import RecommendationProvider from './RecommendationProvider';

const ChannelRecommendation: FC<RecommendationBodyProps> = ({
  location,
  visible = true,
  channel,
}) => {
  return visible ? (
    <RecommendationProvider
      location={location}
      types={['channel']}
      channel={channel}>
      <RecommendationHeader type="channel" location={location} />
      <RecommendationBody
        type="channel"
        location={location}
        channel={channel}
      />
    </RecommendationProvider>
  ) : null;
};

export default ChannelRecommendation;
