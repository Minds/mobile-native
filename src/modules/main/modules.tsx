/**
 * Helper file for aggregating all modules screens, navigators and widgets
 * later to be used by the main navigator, tab navigator, and deep links
 */
import { PathConfigMap } from '@react-navigation/native';
// import * as SettingsComponents from 'modules/settings';
// end-of-modules-import - HYGEN

export const modules = Object.assign(
  {},
  // SettingsComponents,
); // end-of-modules - HYGEN

type Tab = {
  name: string;
  iconName: string;
  group?: string;
};

export const tabs: Tab[] = [
  // {
  //   name: 'Settings',
  //   iconName: 'cards-lock',
  // },
]; // end-of-tab-list - HYGEN

type Config = {
  screens: PathConfigMap<Record<string, never>>;
};

export const config: Config = {
  screens: {
    Main: {
      screens: {
        // Settings: modules.Settings,
      }, // end-of-config-links - HYGEN
    },
  },
};
