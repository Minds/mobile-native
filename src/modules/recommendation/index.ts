export { default as RecommendationProvider } from './Recommendation/RecommendationProvider';
export { default as RecommendationHeader } from './Recommendation/RecommendationHeader';
export { default as RecommendationBody } from './Recommendation/RecommendationBody';
export { default as Recommendation } from './Recommendation/Recommendation';
export { default as ChannelRecommendationItem } from './Recommendation/components/ChannelRecommendationItem';
export * from './Recommendation/types';
