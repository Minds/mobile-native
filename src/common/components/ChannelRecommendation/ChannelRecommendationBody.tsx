import { useNavigation } from '@react-navigation/core';
import { observer } from 'mobx-react';
import React, { FC, useCallback, useState } from 'react';
import { View } from 'react-native';
import UserModel from '~/channel/UserModel';
import Subscribe from '~/channel/v2/buttons/Subscribe';
import { useLegacyStores } from '~/common/hooks/use-stores';
import i18n from '~/common/services/i18n.service';
import { Avatar, B1, B2, Column, Icon, Row, Spacer } from '~/common/ui';
import ThemedStyles from '~/styles/ThemedStyles';
import MPressable from '../MPressable';
import { ChannelRecommendationStore } from './hooks/useChannelRecommendation';
import useChannelRecommendationContext from './hooks/useChannelRecommendationContext';
import { hasVariation } from '../../../../ExperimentsProvider';

interface ChannelRecommendationItemProps {
  channel: UserModel;
  onSubscribed?: (user: UserModel) => void;
}

export const ChannelRecommendationItem: FC<ChannelRecommendationItemProps> = ({
  channel,
  onSubscribed,
}) => {
  const avatar =
    channel && channel.getAvatarSource ? channel.getAvatarSource('medium') : {};
  const navigation = useNavigation<any>();
  const onPress = useCallback(
    () =>
      navigation.push('Channel', {
        guid: channel.guid,
        entity: channel,
      }),
    [navigation, channel],
  );

  const description =
    channel.briefdescription && channel.briefdescription.trim
      ? channel.briefdescription.trim()
      : '';

  return (
    <MPressable onPress={onPress}>
      <Row vertical="S" horizontal="L">
        <Avatar size="tiny" right="M" top="XS" source={avatar} />
        <Column flex align="centerStart" right="L">
          <B1 font="bold">{channel.name}</B1>
          {Boolean(description) && (
            <B2 numberOfLines={2} color="secondary">
              {description}
            </B2>
          )}
          {!!channel.boosted && <BoostedChannelLabel />}
        </Column>
        <Subscribe
          mini
          shouldUpdateFeed={false}
          channel={channel}
          onSubscribed={onSubscribed}
        />
      </Row>
    </MPressable>
  );
};

const BoostedChannelLabel = () => (
  <Row top="XS" align="centerStart">
    <Icon name="boost" size="tiny" right="XS" color="Link" />
    <B2 color="link">{i18n.t('boosts.boostedChannel')}</B2>
  </Row>
);

export interface ChannelRecommendationProps {
  location: string;
  /**
   * use this prop to allow the component to prefetch the data but not render the component
   */
  visible?: boolean;
  /**
   * the channel for which we should get recommendations
   */
  channel?: UserModel;

  recommendationStore?: ChannelRecommendationStore;
}

const ChannelRecommendationBody: FC<ChannelRecommendationProps> = ({
  location,
  visible = true,
  recommendationStore,
}) => {
  const RECOMMANDATIONS_SIZE = hasVariation('mob-4638-boost-v3') ? 4 : 3;
  const [listSize, setListSize] = useState(RECOMMANDATIONS_SIZE);
  const recommendation =
    useChannelRecommendationContext() || recommendationStore;

  const { dismissal } = useLegacyStores();
  const isDismissed = dismissal.isDismissed('channel-recommendation:feed');
  const dismissible = location !== 'channel';

  const shouldRender =
    Boolean(recommendation?.result?.entities.length) &&
    visible &&
    (dismissible ? !isDismissed : true);

  /**
   * When a channel was subscribed, remove it from the list——unless the list is small
   */
  const onSubscribed = useCallback(
    subscribedChannel => {
      if (!recommendation?.result?.entities) {
        return;
      }

      if (recommendation.result.entities.length <= RECOMMANDATIONS_SIZE) {
        return null;
      }

      // LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
      recommendation?.setResult({
        ...recommendation?.result,
        entities: recommendation?.result?.entities.filter(
          suggestion => suggestion.entity_guid !== subscribedChannel.guid,
        ),
      });
      if (listSize === RECOMMANDATIONS_SIZE) {
        setListSize(RECOMMANDATIONS_SIZE + 2);
      }
    },
    [recommendation, listSize],
  );

  if (!shouldRender) {
    return null;
  }

  return (
    <>
      <Spacer bottom="XL">
        {recommendation?.result?.entities.slice(0, listSize).map(suggestion => (
          <ChannelRecommendationItem
            key={suggestion.entity.guid}
            channel={suggestion.entity}
            onSubscribed={onSubscribed}
          />
        ))}
      </Spacer>
      <View style={styles.borderBottom} />
    </>
  );
};

const styles = ThemedStyles.create({
  borderBottom: ['borderBottom6x', 'bcolorBaseBackground'],
});

export default observer(ChannelRecommendationBody);
