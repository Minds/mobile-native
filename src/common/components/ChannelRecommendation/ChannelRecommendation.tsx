import React, { FC } from 'react';
import ChannelRecommendationBody, {
  ChannelRecommendationProps,
} from './ChannelRecommendationBody';
import ChannelRecommendationHeader from './ChannelRecommendationHeader';
import { useChannelRecommendation } from './hooks/useChannelRecommendation';

const ChannelRecommendation: FC<ChannelRecommendationProps> = ({
  location,
  visible = true,
  channel,
}) => {
  const recommendation = useChannelRecommendation(location, channel);
  return visible ? (
    <>
      <ChannelRecommendationHeader
        location={location}
        recommendationStore={recommendation}
      />
      <ChannelRecommendationBody
        location={location}
        channel={channel}
        recommendationStore={recommendation}
      />
    </>
  ) : null;
};

export default ChannelRecommendation;
