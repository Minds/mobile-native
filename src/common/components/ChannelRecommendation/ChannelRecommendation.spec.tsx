import React from 'react';
import ChannelRecommendation from './ChannelRecommendation';

import { shallow } from 'enzyme';
import UserModel from '~/channel/UserModel';
import generateFakeUser from '../../../../__mocks__/fake/channel/UserFactory';
import { ChannelRecommendationItem } from '~/modules/recommendation';

const mockEntity = generateFakeUser();

jest.mock('~/common/hooks/useApiFetch', () => () => ({
  result: {
    status: '',
    entities: [
      {
        entity: mockEntity,
        guid: '',
      },
    ],
  },
}));

jest.mock('~/common/hooks/use-stores', () => ({
  useLegacyStores: () => ({
    recentSubscriptions: {
      list() {
        return [];
      },
    },
    dismissal: {
      isDismissed() {
        return false;
      },
      dismiss() {
        return;
      },
    },
  }),
}));

describe('ChannelRecommendation', () => {
  test('renders correctly', () => {
    const component = shallow(
      <ChannelRecommendation location="fakeLocation" />,
    );
    expect(component).toMatchSnapshot();
  });
});

describe('ChannelRecommendationItem', () => {
  test('renders correctly', () => {
    const componentItem = shallow(
      <ChannelRecommendationItem channel={UserModel.create(mockEntity)} />,
    );
    expect(componentItem).toMatchSnapshot();
  });
});
