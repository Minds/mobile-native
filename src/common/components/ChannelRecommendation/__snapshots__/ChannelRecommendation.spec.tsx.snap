// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelRecommendation renders correctly 1`] = `
<Fragment>
  <Spacer
    vertical="XL"
  >
    <Row
      align="centerBetween"
      bottom="XL"
      horizontal="L"
    >
      <H3>
        Recommended Channels
      </H3>
      <B2
        color="link"
        onPress={[Function]}
      >
        See more
      </B2>
    </Row>
    <ChannelRecommendationItem
      channel={
        Object {
          "attachment_guid": false,
          "blurb": false,
          "custom_data": false,
          "custom_type": false,
          "description": "Congratulations! ",
          "edited": "",
          "guid": "",
          "mature": false,
          "message": "Message",
          "ownerObj": Object {
            "guid": "824853017709780997",
            "subtype": false,
            "time_created": "1522036284",
            "type": "user",
          },
          "owner_guid": "824853017709780997",
          "parent_guid": "838106762591510528",
          "perma_url": false,
          "rowKey": "something",
          "shouldBeBlured": [MockFunction],
          "thumbnail_src": false,
          "time_created": "1522036284",
          "title": "TITLE",
          "type": "activity",
          "wire_totals": Object {
            "tokens": 1000000000000000000,
          },
        }
      }
    />
  </Spacer>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 6,
        },
        Object {
          "borderColor": "#3f4b56",
        },
      ]
    }
  />
</Fragment>
`;

exports[`ChannelRecommendationItem renders correctly 1`] = `
<MPressable
  onPress={[Function]}
>
  <Row
    horizontal="L"
    vertical="S"
  >
    <withSpacer()
      right="M"
      size="tiny"
      source={
        Object {
          "headers": Object {
            "App-Version": "3.8.0",
            "Cache-Control": "no-cache, no-store, must-revalidate",
            "Content-Type": "application/json",
            "Pragma": "no-cache",
            "no-cache": "1",
          },
          "uri": "https://cdn.minds.com/icon//medium/undefined",
        }
      }
      top="XS"
    />
    <Column
      align="centerStart"
      flex={true}
    >
      <B1
        font="bold"
      />
    </Column>
    <Subscribe
      channel={
        Object {
          "allow_comments": true,
          "attachment_guid": false,
          "blocked": undefined,
          "blurb": false,
          "canary": false,
          "confirmEmail": [Function],
          "custom_data": false,
          "custom_type": false,
          "description": "Congratulations! ",
          "disable_autoplay_videos": undefined,
          "edited": "",
          "email_confirmed": false,
          "eth_wallet": "",
          "founder": false,
          "groupsCount": 0,
          "guid": "",
          "impressions": undefined,
          "isOwner": [Function],
          "is_admin": false,
          "last_accepted_tos": 0,
          "mature": false,
          "mature_visibility": false,
          "message": "Message",
          "mode": 0,
          "nsfw": Array [],
          "onchain_booster": 0,
          "ownerObj": Object {
            "guid": "824853017709780997",
            "subtype": false,
            "time_created": "1522036284",
            "type": "user",
          },
          "owner_guid": "824853017709780997",
          "parent_guid": "838106762591510528",
          "pending_subscribe": false,
          "perma_url": false,
          "permissions": Object {},
          "plus": false,
          "plus_method": "tokens",
          "pro": false,
          "pro_method": "tokens",
          "rewards": false,
          "rowKey": "something",
          "shouldBeBlured": [MockFunction],
          "subscribed": undefined,
          "subscriber": false,
          "subscribers_count": undefined,
          "subscriptions_count": 0,
          "tags": Array [],
          "thumbnail_src": false,
          "time_created": "1522036284",
          "time_updated": "",
          "title": "TITLE",
          "type": "activity",
          "urn": "",
          "username": "",
          "verified": false,
          "wire_rewards": undefined,
          "wire_totals": Object {
            "tokens": 1000000000000000000,
          },
        }
      }
      mini={true}
      shouldUpdateFeed={false}
    />
  </Row>
</MPressable>
`;
